### ISSUE TITLE
Building 5.0.0 SNAPSHOTs and using TransportClient in other projects in hard

### ISSUE THREAD

**Issue by dadoonet** [elastic/elasticsearch#20375]
I'm trying to build a SNAPSHOT version of the latest master to be able to integrate it in a Test project.

After cloning the elasticsearch repo, I'm running:

```sh
gradle install
```

It works well.

Then I'm creating another project (https://github.com/dadoonet/legacy-search) where I want to use this SNAPSHOT version I just built.
In my  `pom.xml`, I'm writing:

```xml
<dependency>
    <groupId>org.elasticsearch.client</groupId>
    <artifactId>transport</artifactId>
    <version>5.0.0-alpha6-SNAPSHOT</version>
</dependency>
```

But sadly, some artifacts are not available in the standard `.m2` dir. When I run `mvn install` I'm getting:

```
[WARNING] The POM for org.elasticsearch.plugin:transport-netty3-client:jar:5.0.0-alpha6-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for org.elasticsearch.plugin:reindex-client:jar:5.0.0-alpha6-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for org.elasticsearch.plugin:lang-mustache-client:jar:5.0.0-alpha6-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for org.elasticsearch.plugin:transport-netty4-client:jar:5.0.0-alpha6-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for org.elasticsearch.plugin:percolator-client:jar:5.0.0-alpha6-SNAPSHOT is missing, no dependency information available
```

So basically the `pom.xml` for `org.elasticsearch.client:transport` and `org.elasticsearch:elasticsearch` have been published in the local `.m2` but I'm missing other jars/pom.

For example:

```
$ ls -l ~/.m2/repository/org/elasticsearch/plugin/transport-netty3-client
total 0
drwxr-xr-x  9 dpilato  staff  306 Sep  7 17:28 5.0.0-alpha5
```

We only have "released" versions here.

Is this something we can fix or does it mean we can't really test the transport client until a release or a snapshot has been published on elastic repository or on sonatype?

I believe the same should happen with Gradle unless there is a workaround?

**Comment by rjernst** [comment]
This is a known issue which I am working on fixing. I described the problem in #20164, but to summarize, `gradle install` is the "old" way of doing things. With the changes made for unified release, we should be using `gradle publishToMavenLocal`. However, that currently will not quite work either, due to how we hacked in transport client plugins. I am working on simplifying all this.

**Comment by dadoonet** [comment]
Great news! Thanks Ryan for working on that!
